### ISSUE TITLE
JITServer CHTable WARNING: classInfo for subclass 0x207e500 of 0x1f32c00 is null

### ISSUE THREAD

**Issue by a7ehuo** [eclipse/openj9#7035]
Observed the following warning from `CHTable` (`JITServerPersistentCHTable.cpp`) while running sanity test. 
```
CHTable WARNING: classInfo for subclass 0x207e500 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207e200 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207de00 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207da00 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207d200 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207cf00 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207cb00 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207c700 of 0x1f32c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x207c300 of 0x1f32c00 is null, ignoring
```

It'd be observed in multiple tests, one of which is `cmdLineTester_GCRegressionTests_2`. Narrowed it down to the following sub-command: 
```
Running command: "/root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/bin/java" -Xcompressedrefs -XX:+UseCompressedOops -Xjit -Xgcpolicy:balanced  -XX:+UseJITServer   -Dcom.ibm.tools.attach.enable=no -Xdump:system:events=systhrow,filter=java/lang/OutOfMemoryError -Xmx4m -Xms4m -Xalwaysclassgc -Xdisableexcessivegc   -cp /root/home/ahuo/src/openj9-openjdk-jdk8/openj9/test/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/cmdLineTests/gcRegressionTests/gcRegressionTests.jar com.ibm.tests.garbagecollector.TestClassLoaderMain - - -
```

**Comment by a7ehuo** [comment]
Narrowed the  `CHTable WARNING` to the commit 16fb704. 
16fb704 ==> `CHTable WARNING` happens. 
312391c ==>  No warnings. 

https://github.com/eclipse/openj9/compare/312391c...16fb704

Before the `CHTable WARNING` occurs, there is a message type mismatch. May or may not be related. Will investigate further.

`#JITaaS: Stream message type mismatch: JITServer expected mesasge type 5 received 2`
```
mirrorResolvedJ9Method = 2;
compilationRequest = 5; // type used when client sends remote compilation requests
```

```
#JITaaS: compThreadID=1 did an early abort
#JITaaS: Server received request for stream 00007FADCA3B3310
#JITaaS: compThreadID=1 found clientSessionData=00007FADCA3B34E0 for clientUID=18043674595488648783 seqNo=139
#JITaaS: Server will process a list of 4972 unloaded classes for clientUID 18043674595488648783
#JITaaS: Stream interrupted by JITClient while compThreadID=1 was reading the compilation request: Compilation interrupted at JITClient's request
#JITaaS: compThreadID=1 did an early abort for clientUID=18043674595488648783 seqNo=139
#JITaaS: compThreadID=1 found clientSessionData=00007FADCA3B34E0 for clientUID=18043674595488648783 seqNo=140
#JITaaS: Server will process a list of 1416 unloaded classes for clientUID 18043674595488648783
#FAILURE:  <EARLY TRANSLATION FAILURE: compilation aborted>
#JITaaS: Stream message type mismatch: JITServer expected mesasge type 5 received 2
#JITaaS: compThreadID=1 did an early abort
#JITaaS: compThreadID=1 found clientSessionData=00007FADCA3B34E0 for clientUID=18043674595488648783 seqNo=141
#JITaaS: Server will process a list of 0 unloaded classes for clientUID 18043674595488648783
CHTable WARNING: classInfo for subclass 0x2254d00 of 0x2138c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x2254a00 of 0x2138c00 is null, ignoring
CHTable WARNING: classInfo for subclass 0x2254600 of 0x2138c00 is null, ignoring
...
```

**Comment by mpirvu** [comment]
I was thinking about this I believe the problem stems from not applying CHTable updates when the communication between server and client is broken very early on.
The CHTable updates need to be processed in order and therefore the client gets a "sequence number" when computing the list of updates that needs to be sent to the server:
```
1. acquire sequencing lock
2.    copy the global list of CHTable updates to a private list and clear the global list
3.    increment the seqNo
4. release sequencing lock
5. send compilation request
6. handle messages from server
// Note that typically the first message received from the server is a request for CHTable updates
```
The server makes sure that it applies the CHTable updates in order:
```
1. read compilation request including the seqNo set by the client
2. retrieve client session
3. acquire sequencing lock
4.    if (seqNo > expectedSeqNo) waitForMyTurn
5.    else if (seqNo < expectedSeqNo) throw StreamOOO;
6. release sequencing lock
7. process list of unloaded classes
8. initialize CHTable if needed (only for a never seen before client)
9. send CHTable_getClassInfoUpdates
10. apply CHTable updates
```
As shown above, if the client has some CHTable updates to send, the server MUST receive them because the client will not attempt to send the same list twice. However, it's possible that the server throws errors either because
(a) The compilation request at step (1) is never received
(b) The client interrupts the compilation very early and the server instead of receiving the list of CHTable updates at step (9), receives an interruption message.

Case (b) could be handled easily either by the client sending its CHTable list together with the interruption message, or always accepting the CHTable update request and postponing to send the interruption message until the next query from the server.

Case (a) is much more difficult. One way to deal with it is to destroy the entire CHTable and start over, but I am not sure it's safe to destroy the CHTable at any random point in time, because other compilation threads for the same client may depend on it.

**Comment by mpirvu** [comment]
I attempted to reproduce this issue with `cmdLineTester_GCRegressionTests_2`, but the master branch code fails this test without hitting the FATAL_ASSERT.
```
JVM_OPTIONS: -Xcompressedrefs -XX:+UseCompressedOops -Xjit -Xgcpolicy:balanced -XX:+UseJITServer
{ itercnt=1; \
mkdir -p "/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../TestConfig/test_output_1573245565120/cmdLineTester_GCRegressionTests_2"; \
cd "/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../TestConfig/test_output_1573245565120/cmdLineTester_GCRegressionTests_2"; \
"/home/mpirvu/FullVM/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/bin/java" -Xcompressedrefs -XX:+UseCompressedOops -Xjit -Xgcpolicy:balanced -XX:+UseJITServer  -DTESTSJARPATH="/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../../jvmtest/functional/cmdLineTests/gcRegressionTests/gcRegressionTests.jar" -DRESJAR="/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../../jvmtest/functional/cmdline_options_testresources/cmdlinetestresources.jar" \
        -DEXE='"/home/mpirvu/FullVM/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/bin/java" -Xcompressedrefs -XX:+UseCompressedOops -Xjit -Xgcpolicy:balanced -XX:+UseJITServer ' -Xint -jar "/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../../jvmtest/functional/cmdline_options_tester/cmdlinetester.jar" -config "/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../../jvmtest/functional/cmdLineTests/gcRegressionTests/gcRegressionTests.xml" \
        -verbose -explainExcludes -xids all,linux_x86-64_cr, -plats all,linux_x86-64_cr, -xlist "/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../../jvmtest/functional/cmdLineTests/gcRegressionTests/gcRegressionTests_excludes.xml" -nonZeroExitWhenError; \
        if [ $? -eq 0 ] ; then echo ""; echo "cmdLineTester_GCRegressionTests_2""_PASSED"; echo ""; cd /home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/..;  else echo ""; echo "cmdLineTester_GCRegressionTests_2""_FAILED"; echo ""; fi; } 2>&1 | tee -a "/home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../TestConfig/test_output_1573245565120/TestTargetResult";
Error: Unable to access jarfile /home/mpirvu/FullVM/openj9-openjdk-jdk8/openj9/test/TestConfig/../../jvmtest/functional/cmdline_options_tester/cmdlinetester.jar

cmdLineTester_GCRegressionTests_2_FAILED
```
That `jvmtest` directory does not exist. Maybe I have some stale version of the tests

**Comment by mpirvu** [comment]
I was missing ant-contrib

**Comment by a7ehuo** [comment]
Tried the local sanity test with the `master` branch and no longer observe the `CHTable WARNING:`. 

OpenJ9 build: The` master` branch f12324a + the runtime assumption fix #7948.
```
===========================
JIT: using build "Dec  4 2019 08:57:23" 
JIT level: 0ad337e
openjdk version "1.8.0_242-internal"
OpenJDK Runtime Environment (build 1.8.0_242-internal-root_2019_12_04_08_49-b00)
Eclipse OpenJ9 VM (build marius-fix-runtime-assumption-0ad337e, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20191204_000000 (JIT enabled, AOT enabled)
OpenJ9   - 0ad337e
OMR      - f19cb88
JCL      - 8514ba3 based on jdk8u242-b02)
===========================
```
